File: src/test/java/net/pterodactylus/sone/web/AboutPageTest.java
package net.pterodactylus.sone.web;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import org.junit.Test;

/**
 * Unit test for {@link AboutPage}.
 *
 * @author <a href="mailto:bombe@pterodactylus.net">David ‘Bombe’ Roden</a>
 */
public class AboutPageTest extends WebPageTest {

	private final String version = "0.1.2";
	private final int year = 1234;
	private final String homepage = "home://page";
	private final AboutPage page = new AboutPage(template, webInterface, version, year, homepage);

	@Test
	public void pageReturnsCorrectPath() {
		assertThat(page.getPath(), is("about.html"));
	}

	@Test
	public void pageSetsCorrectVersionInTemplateContext() throws Exception {
		page.processTemplate(freenetRequest, templateContext);
		assertThat(templateContext.get("version"), is((Object) version));
	}

	@Test
	public void pageSetsCorrectHomepageInTemplateContext() throws Exception {
		page.processTemplate(freenetRequest, templateContext);
		assertThat(templateContext.get("homepage"), is((Object) homepage));
	}

	@Test
	public void pageSetsCorrectYearInTemplateContext() throws Exception {
		page.processTemplate(freenetRequest, templateContext);
		assertThat(templateContext.get("year"), is((Object) year));
	}

}

File: src/test/kotlin/net/pterodactylus/sone/web/AboutPageTest.kt
package net.pterodactylus.sone.web

import net.pterodactylus.sone.main.SonePlugin.PluginHomepage
import net.pterodactylus.sone.main.SonePlugin.PluginVersion
import net.pterodactylus.sone.main.SonePlugin.PluginYear
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.junit.Test

/**
 * Unit test for [AboutPage].
 */
class AboutPageTest: WebPageTest() {

	private val version = "0.1.2"
	private val year = 1234
	private val homepage = "home://page"
	private val page = AboutPage(template, webInterface, PluginVersion(version), PluginYear(year), PluginHomepage(homepage))

	@Test
	fun `page returns correct path`() {
		assertThat(page.path, equalTo("about.html"))
	}

	@Test
	fun `page does not require login`() {
		assertThat(page.requiresLogin(), equalTo(false))
	}

	@Test
	fun `page sets correct version in template context`() {
		page.processTemplate(freenetRequest, templateContext)
		assertThat(templateContext["version"], equalTo<Any>(version))
	}

	@Test
	fun `page sets correct homepage in template context`() {
		page.processTemplate(freenetRequest, templateContext)
		assertThat(templateContext["homepage"], equalTo<Any>(homepage))
	}

	@Test
	fun `page sets correct year in template context`() {
		page.processTemplate(freenetRequest, templateContext)
		assertThat(templateContext["year"], equalTo<Any>(year))
	}

}
